lib/core/http/fetch-wrapper.ts

```diff
@@ -0,0 +1,58 @@
+import { HttpError } from "@lib/core/http/error";
+
+async function http<T>(path: string, config: RequestInit): Promise<T> {
+  const request = new Request(path, config);
+  const response: Response = await fetch(request);
+
+  if (!response.ok) {
+    const err = HttpError.fromRequest(request, response);
+    throw err;
+  }
+  // may error if there is no body, return empty array
+  return response.json().catch(() => ({}));
+}
+
+export async function get<T>(path: string, config?: RequestInit): Promise<T> {
+  const init = { method: "GET", ...config };
+  return await http<T>(path, init);
+}
+
+export async function post<T, U>(path: string, body: T, config?: RequestInit): Promise<U> {
+  const init = {
+    method: "POST",
+    headers: { "Content-Type": "application/json" },
+    body: JSON.stringify(body),
+    ...config,
+  };
+  return await http<U>(path, init);
+}
```

Comment:
You can name variables in generics too :)

```suggestion
export async function post<TInput, TOutput>(path: string, body: TInput, config?: RequestInit): Promise<TOutput> {
  const init = {
    method: "POST",
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify(body),
    ...config,
  };
  return await http<TOutput>(path, init);
}
```

Reply:
Do you think this is something worth doing?

Reply:
This one was nice-to-have comment, don't worry about doing it if you don't wna